### ISSUE TITLE
This is a Meson bug and should be reported!

### ISSUE THREAD

**Issue by AESASH** [mesonbuild/meson#12995]
**Describe the bug**
When building with meson, I get the following error: 
```
FileNotFoundError: [Errno 2] No such file or directory: '/home/qt/work/install/include'
```

And as you see the path does not include the name of the user under "home"

Commands I used to build

```
export PKG_CONFIG_PATH=~/Qt/6.6.2/gcc_64/lib/pkgconfig:$PKG_CONFIG_PATH
meson build
```

ERROR BELOW:

```
The Meson build system
Version: 0.61.2
Source dir: /home/aes/Desktop/aes/libportal
Build dir: /home/aes/Desktop/aes/libportal/build
Build type: native build
Project name: libportal
Project version: 0.7.1
C compiler for the host machine: ccache cc (gcc 11.4.0 "cc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0")
C linker for the host machine: cc ld.bfd 2.38
Host machine cpu family: x86_64
Host machine cpu: x86_64
Compiler for C supports arguments -Wno-unused-parameter: YES 
Compiler for C supports arguments -Wno-missing-field-initializers: YES 
Compiler for C supports arguments -fvisibility=hidden: YES 
Has header "sys/vfs.h" : YES 
Configuring config.h using configuration
Program glib-mkenums found: YES (/usr/bin/glib-mkenums)
Program glib-mkenums found: YES (/usr/bin/glib-mkenums)
Found pkg-config: /usr/bin/pkg-config (0.29.2)
Run-time dependency gio-2.0 found: YES 2.72.4
Run-time dependency gio-unix-2.0 found: YES 2.72.4
Run-time dependency gobject-introspection-1.0 found: YES 1.72.0
Dependency gobject-introspection-1.0 found: YES 1.72.0 (cached)
Program g-ir-scanner found: YES (/usr/bin/g-ir-scanner)
Dependency gobject-introspection-1.0 found: YES 1.72.0 (cached)
Program g-ir-compiler found: YES (/usr/bin/g-ir-compiler)
Program vapigen found: YES (/usr/bin/vapigen)
WARNING: Unknown keyword arguments in target libportal.vapi: packages
Run-time dependency gtk+-3.0 found: YES 3.24.33
Program vapigen found: YES (/usr/bin/vapigen)
WARNING: Unknown keyword arguments in target libportal-gtk3.vapi: packages
Run-time dependency gtk4 found: YES 4.6.9
Program vapigen found: YES (/usr/bin/vapigen)
WARNING: Unknown keyword arguments in target libportal-gtk4.vapi: packages
C++ compiler for the host machine: ccache c++ (gcc 11.4.0 "c++ (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0")
C++ linker for the host machine: c++ ld.bfd 2.38
Run-time dependency qt5 (modules: Core, Gui, Widgets, X11Extras) found: YES 5.15.3 (pkg-config)
libportal/meson.build:281: WARNING: add_languages is missing native:, assuming languages are wanted for both host and build.
C++ compiler for the host machine: ccache c++ (gcc 11.4.0 "c++ (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0")
C++ linker for the host machine: c++ ld.bfd 2.38
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/mesonbuild/mesonmain.py", line 146, in run
    return options.run_func(options)
  File "/usr/lib/python3/dist-packages/mesonbuild/msetup.py", line 294, in run
    app.generate()
  File "/usr/lib/python3/dist-packages/mesonbuild/msetup.py", line 185, in generate
    self._generate(env)
  File "/usr/lib/python3/dist-packages/mesonbuild/msetup.py", line 229, in _generate
    intr.run()
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreter/interpreter.py", line 2549, in run
    super().run()
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase/interpreterbase.py", line 148, in run
    self.evaluate_codeblock(self.ast, start=1)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase/interpreterbase.py", line 173, in evaluate_codeblock
    raise e
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase/interpreterbase.py", line 166, in evaluate_codeblock
    self.evaluate_statement(cur)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase/interpreterbase.py", line 179, in evaluate_statement
    return self.function_call(cur)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase/interpreterbase.py", line 454, in function_call
    res = func(node, func_args, kwargs)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase/decorators.py", line 276, in wrapper
    return f(*nargs, **wrapped_kwargs)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase/decorators.py", line 573, in wrapper
    return f(*wrapped_args, **wrapped_kwargs)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreter/interpreter.py", line 2039, in func_subdir
    self.evaluate_codeblock(codeblock)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase/interpreterbase.py", line 173, in evaluate_codeblock
    raise e
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase/interpreterbase.py", line 166, in evaluate_codeblock
    self.evaluate_statement(cur)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase/interpreterbase.py", line 189, in evaluate_statement
    return self.evaluate_if(cur)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase/interpreterbase.py", line 272, in evaluate_if
    self.evaluate_codeblock(i.block)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase/interpreterbase.py", line 173, in evaluate_codeblock
    raise e
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase/interpreterbase.py", line 166, in evaluate_codeblock
    self.evaluate_statement(cur)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase/interpreterbase.py", line 181, in evaluate_statement
    self.assignment(cur)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase/interpreterbase.py", line 566, in assignment
    value = self.evaluate_statement(node.value)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase/interpreterbase.py", line 179, in evaluate_statement
    return self.function_call(cur)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase/interpreterbase.py", line 454, in function_call
    res = func(node, func_args, kwargs)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase/decorators.py", line 768, in wrapped
    return f(*wrapped_args, **wrapped_kwargs)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase/decorators.py", line 768, in wrapped
    return f(*wrapped_args, **wrapped_kwargs)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase/decorators.py", line 768, in wrapped
    return f(*wrapped_args, **wrapped_kwargs)
  [Previous line repeated 5 more times]
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase/decorators.py", line 96, in wrapped
    ret = f(*wrapped_args, **wrapped_kwargs)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase/decorators.py", line 115, in wrapped
    return f(*wrapped_args, **wrapped_kwargs)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase/decorators.py", line 276, in wrapper
    return f(*nargs, **wrapped_kwargs)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreter/interpreter.py", line 1572, in func_dependency
    d = df.lookup(kwargs)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreter/dependencyfallbacks.py", line 347, in lookup
    dep = func(kwargs, func_args, func_kwargs)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreter/dependencyfallbacks.py", line 90, in _do_dependency
    dep = dependencies.find_external_dependency(name, self.environment, kwargs)
  File "/usr/lib/python3/dist-packages/mesonbuild/dependencies/detect.py", line 114, in find_external_dependency
    d = c()
  File "/usr/lib/python3/dist-packages/mesonbuild/dependencies/factory.py", line 99, in <lambda>
    DependencyMethods.PKGCONFIG:      lambda env, kwargs: pkgconfig_class(pkgconfig_name or name, env, kwargs),
  File "/usr/lib/python3/dist-packages/mesonbuild/dependencies/qt.py", line 193, in __init__
    mod_private_inc = _qt_get_private_includes(mod_private_dir, m, mod.version)
  File "/usr/lib/python3/dist-packages/mesonbuild/dependencies/qt.py", line 54, in _qt_get_private_includes
    dirs = [filename for filename in os.listdir(mod_inc_dir)
FileNotFoundError: [Errno 2] No such file or directory: '/home/qt/work/install/include'

libportal/meson.build:282:3: ERROR: Unhandled python exception

    This is a Meson bug and should be reported!
```

**To Reproduce**
Just build libportal "https://github.com/flatpak/libportal" on Ubuntu 22.04, this happened after I have installed Qt6 dependency and did "meson build" . 
You will see all the meson.build files you ask for on the Repo page

**Expected behavior**
I expected the build to succeed

**system parameters**
* Plain native installation
* Ubuntu 22.04
* pythin 3.10.12
* meson 0.61.2
* ninja 1.10.1

**Comment by eli-schwartz** [comment]
Thanks for the report. This was fixed in #12233 and backported to meson 1.2.2

Unfortunately Ubuntu LTS packages an old version of meson. This means you don't get new meson bugs, but you also don't get new fixes for old bugs. :(

There are a few different ways to get an updated version of meson, described at https://mesonbuild.com/Getting-meson.html
If you don't need to run `sudo meson install` then the easiest way is probably `pip install --user meson`.

**Comment by eli-schwartz** [comment]
Duplicate of #12214
